### PR TITLE
utils: fix SIMD boundary for JSON string encoding

### PR DESCRIPTION
**Pull request by cosmo0920**
## Summary

Fix JSON string corruption caused by SIMD reads crossing the declared end of long strings after processing multibyte UTF-8 characters.

Affected strings could receive 1–15 bytes of adjacent memory on 16-byte SIMD platforms, or fewer bytes with narrower SIMD implementations. Loki JSON output with `remove_keys` made the issue especially reproducible and could expose fragments of other log records.

## Root cause

The JSON string writers rounded `str_len` down to a SIMD-aligned boundary and used `i < vlen` to decide whether another vector could be processed.

After scalar processing of a 2–4-byte UTF-8 sequence, `i` may no longer be SIMD-aligned. Consequently, `i < vlen` did not guarantee that the complete vector remained within `str_len`, allowing bytes past the logical end of the string to be read and copied.

## Changes

- Require the entire SIMD vector to fit within the remaining input:

  ```c
  i + inst_len <= str_len
  ```

- Apply the correction to both escaped and raw JSON string writers.
- Add an internal regression test with poisoned bytes after the declared input boundary.
- Cover SIMD-width, 24 KiB, and 36 KiB payload sizes.
- Add Loki runtime coverage using JSON line formatting and `remove_keys`.
- Add an end-to-end integration test that sends 200 approximately 40 KiB Unicode records through tail → Loki and compares every decoded message with its original value.

## Compatibility

No configuration or API changes. The only behavioral change is that bytes beyond the declared string length are no longer included in encoded output.

## Verification

Recorded in the referenced task, “Fix JSON string corruption”:

- `cmake --build build --target flb-it-utils -j8` — passed
- `./build/bin/flb-it-utils write_str_simd_boundary` — passed
- `ctest --test-dir build -R '^flb-it-utils$' --output-on-failure` — passed
- macOS Leaks reported no leaks, although macOS restricted full process inspection

The latest Loki runtime and new Python integration tests still need final functional and macOS Leaks runs before submitting the PR.

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:

- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change

- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.

- [ ] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [ ] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**

- [ ] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of long escaped and raw strings, including multibyte Unicode content.
  * Prevented data corruption when string lengths fall near SIMD processing boundaries.
  * Preserved complete JSON messages and formatting for large Loki payloads.

* **Tests**
  * Added coverage for long Unicode messages, boundary-length payloads, and Loki integration output.

### CODE REVIEW

File: tests/integration/scenarios/out_loki/tests/test_out_loki_001.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
+import json
```
[the remark this answers is not in the document]
Comment (cosmo0920) [reply tests/integration/scenarios/out_loki/tests/test_out_loki_001.py:1]: The main fix of this PR is only this commit: https://github.com/fluent/fluent-bit/pull/12285/commits/c323e9e58b51667fa7636763e0eaaf89f207848d.

Other commits are just adding their additional test cases so we can only use `utils:` as main topic prefix in this PR.
